robottelo/hosts.py

```diff
@@ -641,16 +641,24 @@ def update_known_hosts(self, ssh_key_name, host, user=None):
                 f'Failed to put hostname in ssh known_hosts files:\n{result.stderr}'
             )
 
-    def configure_puppet(self, rhel_repo=None, proxy_hostname=None):
+    def configure_puppet(self, proxy_hostname=None):
         """Configures puppet on the virtual machine/Host.
         :param proxy_hostname: external capsule hostname
-        :param rhel_repo: dictionary mapping repo name to repo baseurl.
         :return: None.
+        :raises robottelo.hosts.ContentHostError: If installation or configuration fails.
         """
         if proxy_hostname is None:
             proxy_hostname = settings.server.hostname
 
-        self.create_custom_repos(**rhel_repo)
+        self.create_custom_repos(
+            **{'sat-client': settings.repos['SATCLIENT_REPO'][f'RHEL{self.os_version.major}']}
```

Comment:
This is a bit clunky to create the dict and expand it, instead of just passing kwargs directly.